config/confighttp/confighttp.go

```diff
@@ -180,7 +180,7 @@ func (hcs *ClientConfig) ToClient(ctx context.Context, host component.Host, sett
 			return nil, errors.New("extensions configuration not found")
 		}
 
-		httpCustomAuthRoundTripper, aerr := hcs.Auth.GetClientAuthenticator(ext)
+		httpCustomAuthRoundTripper, aerr := hcs.Auth.GetClientAuthenticatorContext(context.Background(), ext)
```

Comment:
Same comment as the one for gRPC: this function receives a context already, perhaps it can reuse it?